app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -840,9 +840,11 @@ open class BrowserActivity : DuckDuckGoActivity() {
         val fragment = duckAiFragment
         if (fragment?.isVisible == true) {
             animateDuckAiFragmentOut {
-                val transaction = supportFragmentManager.beginTransaction()
-                transaction.hide(fragment)
-                transaction.commit()
+                if (!supportFragmentManager.isStateSaved) {
```

Comment:
I'm not sure how exactly that scenario would play out, but shouldn't this be checked before we run any logic in the function? Otherwise, if a state is restored with fragment visible, our local `isDuckChatVisible` would be set to `false` already.

Reply:
It needs to be checked when the animation completes, if we check it before, the state may have changed by then. Not sure how to deal with that inconsistency.

Reply:
Right. Anyway, looks like a quick way to prevent the crash and worst case we'll have a one time issue with keyboard pop-up which will resolve itself either way, so that's not a big deal.